File: 2996-smallest-missing-integer-greater-than-sequential-prefix-sum/2996-smallest-missing-integer-greater-than-sequential-prefix-sum.cpp
class Solution {
public:
    int missingInteger(vector<int>& nums) {
        int n=nums.size();
        unordered_map<int,int> ans;
        for(int i=0;i<n;i++){
            ans[nums[i]]=i;
        }
        int sum=nums[0];
        if(nums.size()==1){
            return nums[0]+1;
        }
        else{
            int j=1;
            for(int i=0;i<n;i++){
                if(j>n-1){
                    break;
                }
                if (j>=i && nums[j]==nums[j-1]+1){
                    sum+=nums[j];
                    j++;
                }
                else if(ans.find(sum)!=ans.end()){
                    int next=sum+1;
                    sum=next;
                }
                else if(ans.find(sum)==ans.end()){
                    break;
                }
            }
            
        }
        return sum;
        
    }
};

File: 2996-smallest-missing-integer-greater-than-sequential-prefix-sum/2996-smallest-missing-integer-greater-than-sequential-prefix-sum.java
class Solution {
    public int missingInteger(int[] nums) {
        int n=nums.length;
        List<Integer> lst=new ArrayList<>();
        for(int i=0;i<n;i++){
            lst.add(nums[i]);
        }
        int j=1;
        int ans=0;
        int sum=lst.get(0);
        if(n==1){
            return lst.get(0)+1;
        }
        for(int i=0;i<n;i++){
            
            if(j> n-1) break;
            if(j>=i && nums[j]==nums[j-1]+1){
                sum+=nums[j];
                j++;
                

            }
            else if(!lst.contains(sum)){
                break;
            }
            else if(lst.contains(sum)){
                int next1=sum+1;
                sum=next1;
            }
            
        }
        return sum;
    }
}